### ISSUE TITLE
`posTwice` `testNonCyclic` fails if `dotc.CompilationUnit` does not import `Types.Type`

### ISSUE THREAD

**Issue by nicolasstucki** [lampepfl/dotty#3383]
If we remove the unused import `import dotty.tools.dotc.core.Types.Type` in `dotty.tools.dotc.CompilationUnit` the `testNonCyclic` fails in `posTwice`.

```scala
-- [E006] Unbound Identifier Error: ../compiler/src/dotty/tools/dotc/ast/Trees.scala:817:42 
817 |  abstract class Instance[T >: Untyped <: Type] extends DotClass { inst =>
    |                                          ^^^^
    |                                          not found: type Type
```

Note that the error manifests itself in another class.

**Comment by nicolasstucki** [comment]
#3778 showed that this issue is not completely fixed. Now `testNonCyclic` passes but `testIssue34` fails (http://dotty-ci.epfl.ch/lampepfl/dotty/2372/4).

**Comment by nicolasstucki** [comment]
To consider this issue as fixed we should include 354e2ebc4216da9eb8eb1ea845fb080592432368.
